aptos-move/framework/aptos-framework/sources/aptos_account.move

```diff
@@ -22,13 +47,37 @@ module aptos_framework::aptos_account {
         coin::register<AptosCoin>(&signer);
     }
 
+    /// Convenient function to transfer APT to a recipient account that might not exist.
+    /// This would create the recipient account first, which also registers it to receive APT, before transferring.
     public entry fun transfer(source: &signer, to: address, amount: u64) {
         if (!account::exists_at(to)) {
             create_account(to)
         };
         coin::transfer<AptosCoin>(source, to, amount)
     }
 
+    /// Convenient function to transfer a custom CoinType to a recipient account that might not exist.
+    /// This would create the recipient account first and register it to receive the CoinType, before transferring.
+    public entry fun transfer_coins<CoinType>(from: &signer, to: address, amount: u64) acquires DirectTransferConfig {
+        deposit_coins(to, coin::withdraw<CoinType>(from, amount));
+    }
+
+    /// Convenient function to deposit a custom CoinType into a recipient account that might not exist.
+    /// This would create the recipient account first and register it to receive the CoinType, before transferring.
+    public entry fun deposit_coins<CoinType>(to: address, coins: Coin<CoinType>) acquires DirectTransferConfig {
```

Comment:
do we want a name showing this is more than just depositing coins? For example, initialize_account_with_coin_deposit ? I am bad at naming... just throw some idea here

Reply:
I think deposit_coins is fine. This is meant to initialize behind the scene if necessary, similar to aptos_account::transfer, which can create the account if it doesn't exist behind the scene.